pvlib/ivtools.py

```diff
@@ -0,0 +1,262 @@
+
+
+def fit_sde_sandia(v, i, v_oc, i_sc, v_mp, i_mp, vlim=0.2, ilim=0.1):
+    """ Fits the single diode equation to an IV curve.
+
+    If fitting fails, returns NaN in each parameter.
+
+    Parameters
+    ----------
+    v : numeric
+        Voltage at each point on the IV curve, from 0 to v_oc
+
+    i : numeric
+        Current at each point on the IV curve, from i_sc to 0
+
+    v_oc : float
+        Open circuit voltage
+
+    i_sc : float
+        Short circuit current
+
+    v_mp : float
+        Voltage at maximum power point
+
+    i_mp : float
+        Current at maximum power point
+
+    vlim : float, default 0.2
+        defines linear portion of IV curve i.e. V <= vlim * v_oc
+
+    ilim : float, default 0.1
+        defines exponential portion of IV curve i.e. I > ilim * i_sc
+
+    Returns
+    -------
+    tuple of the following elements:
+
+        photocurrent : float
+            photocurrent, A
+
+        saturation_current : float
+            dark (saturation) current, A
+
+        resistance_shunt : float
+            shunt (parallel) resistance, ohm
+
+        resistance_series : float
+            series resistance, ohm
+
+        nNsVth : float
+            product of diode (ideality) factor n (unitless) x number of
+            cells in series Ns (unitless) x cell thermal voltage Vth (V), V
+
+    Notes
+    -----
+    :py:func:`fit_sde_sandia` obtains values for the five parameters for the
+    single diode equation [1]
+
+    .. math::
+
+        I = IL - I0*[exp((V+I*Rs)/(nNsVth))-1] - (V + I*Rs)/Rsh
+
+    :py:func:`pvsystem.singlediode` for definition of the parameters.
+
+    The fitting method proceeds in four steps:
+        1) simplify the single diode equation
+
+    .. math::
+
+        I = IL - I0*exp((V+I*Rs)/(nNsVth)) - (V + I*Rs)/Rsh
+
+        2) replace Rsh = 1/Gp and re-arrange
+
+    .. math::
+
+        I = IL/(1+Gp*Rs) - (Gp*V)/(1+Gp*Rs) -
+          I0/(1+Gp*Rs)*exp((V+I*Rs)/(nNsVth))
+
+        3) fit the linear portion of the IV curve V <= vlim * v_oc
+
+    .. math::
+
+        I ~ IL/(1+Gp*Rs) - (Gp*V)/(1+Gp*Rs) = beta0 + beta1*V
+
+        4) fit the exponential portion of the IV curve I > ilim * i_sc
+
+    .. math::
+
+        log(beta0 - beta1*V - I) ~ log((I0)/(1+Gp*Rs)) + (V)/(nNsVth) +
+        (Rs*I)/(nNsVth) = beta2 + beta3*V + beta4*I
+
+    Values for ``IL, I0, Rs, Rsh,`` and ``nNsVth`` are calculated from the
+    regression coefficents beta0, beta1, beta3 and beta4.
+
+    References
+    ----------
+    [1] S.R. Wenham, M.A. Green, M.E. Watt, "Applied Photovoltaics" ISBN
+    0 86758 909 4
+    [2] C. B. Jones, C. W. Hansen, Single Diode Parameter Extraction from
+    In-Field Photovoltaic I-V Curves on a Single Board Computer, 46th IEEE
+    Photovoltaic Specialist Conference, Chicago, IL, 2019
+    """
+
+    # Find beta0 and beta1 from linear portion of the IV curve
+    beta0, beta1 = _find_beta0_beta1(v, i, vlim, v_oc)
```

Comment:
I would prefer nomenclature related to the task at hand rather than greek.

Reply:
So would I.  But the beta coefficients don't have ready interpretation as do the five SDE parameters.  Open to suggestions.